app/models/team.rb

```diff
@@ -560,6 +560,20 @@ def filter_by_keywords(query, filters, type = 'FactCheck')
     query.where(Arel.sql("#{tsvector} @@ #{tsquery}"))
   end
 
+  def search_for_similar_articles(query)
+    # query expected to be text
+    result_ids = Bot::Smooch.search_for_similar_published_fact_checks_no_cache('text', query, [self.id]).map(&:id)
+    items = []
+    unless result_ids.blank?
+      # I depend on FactCheck to filter result instead of report_design
+      items = FactCheck.where(report_status: 'published').joins(:claim_description)
+      .joins("INNER JOIN project_medias pm ON pm.id = claim_descriptions.project_media_id ")
```

Comment:
When possible, it's more clear to use `ActiveRecord` syntax. So, I suggest:

`joins(claim_description: :project_media)`.

Instead of:

`.joins("INNER JOIN project_medias pm ON pm.id = claim_descriptions.project_media_id ")`.

Reply:
Done